lib/compilers/haskell.js

```diff
@@ -1,10 +1,43 @@
 var Compile = require('../base-compiler');
+var logger = require('../../lib/logger').logger;
+var child_process = require('child_process');
+var path = require('path');
+var _ = require('underscore-node');
 
 function compileHaskell(info, env) {
     var compiler = new Compile(info, env);
     compiler.optionsForFilter = function (filters, outputFilename, userOptions) {
-        return ['-S', '-g', '-o', this.filename(outputFilename)];
+
+        usedStg = _.any(userOptions, function(opt) {
+            return opt.indexOf("-ddump-stg") > -1;
+        });
+
+        logger.warn("usedStg: ", usedStg);
+
+        if (usedStg) {
+            return ["-ddump-stg", "-ddump-to-file", "-o", this.filename(outputFilename)];
+        } else {
+            return ['-S', '-g', '-o', this.filename(outputFilename)];
+        }
     };
+
+    compiler.handlePostProcessResult = function (result, postResult, outputFilename) {
+        if (usedStg) {
+            logger.info("handlePostProcessResult::usedStg");
```

Comment:
Can you either turn this logger.infos into debugs if they're useful or remove them? It would generate too much output on each compilation